src/buffer.rs

```diff
@@ -239,3 +313,101 @@ pub enum IQMatrix {
     /// Abstraction over `VAIQMatrixBufferHEVC`
     HEVC(hevc::IQMatrixBufferHEVC),
 }
+
+pub enum EncSequenceParameter {
+    H264(h264::EncSequenceParameterBufferH264),
+}
+
+pub enum EncPictureParameter {
+    H264(h264::EncPictureParameterBufferH264),
+}
+
+pub enum EncSliceParameter {
+    H264(h264::EncSliceParameterBufferH264),
+}
+
+pub enum EncMacroblockParameterBuffer {
+    H264(h264::EncMacroblockParameterBufferH264),
+}
+
+pub enum EncCodedBuffer {
+    UndefinedStorage(usize),
+    LocalStorage(Vec<u8>), // This one does not work :C
+}
+
+pub struct CodedBufferSegment<'a> {
+    // pub size: u32,
+    pub bit_offset: u32,
+    pub status: u32,
+    //pub buf: *mut ::std::os::raw::c_void,
+
+    pub buf: &'a [u8],
+}
+
+/// Helper to access segments of mapped coded buffer
+pub struct CodedBuffer<'a> {
+    buffer: Option<Buffer>,
```

Comment:
Why is `Option` used here?

See the comments below about having `PictureSync` here instead (`PictureSync` itself, not `Option<PictureSync>` either).